libraries/Microsoft.Bot.Builder/BotServer.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Servers;
+using Microsoft.Bot.Builder.Middleware;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder
+{
+    public abstract class BotServer
```

Comment:
shouldn't the class name be "AdapterBase"?

Reply:
That would also help have a nicer name for the current "BotFrameworkBotserver".

Reply:
Yeah, BotFrameworkBotServer sucks pretty bad